webpack/assets/javascripts/react_app/common/MountingService.js

```diff
@@ -18,11 +18,18 @@ document.addEventListener('page:before-unload', () => {
   }
 });
 
-export function mount(component, selector, data, flattenData = false) {
+export function mount(
+  component,
+  selector,
+  data,
+  metadata,
+  flattenData = false,
+  context = false
+) {
```

Comment:
can this function receive an object as an argument instead?
or maybe even: `mount(component, selector, data, options)` for not breaking existing implementations.